.github/workflows/pull-request.yml

```diff
@@ -37,30 +37,31 @@ jobs:
       - name: Check
         run: |
           ./pepsi clippy --workspace
-  check-cargo-lock:
-    name: Check Cargo.lock
-    strategy:
-      matrix:
-        path:
-          [
-            .,
-            tools/aliveness,
-            tools/hula,
-          ]
-    runs-on:
-      - self-hosted
-      - v3
-    container:
-      image: 134.28.57.223:5000/hulk
-      options: --user=1000:1000
-    steps:
-      - uses: actions/checkout@v3
-        with:
-          lfs: true
-      - name: Check
-        run: |
-          cd ${{ matrix.path }}
-          cargo update --frozen --workspace
+  # Broken due to serde crates.io patching:
+  # check-cargo-lock:
+  #   name: Check Cargo.lock
+  #   strategy:
+  #     matrix:
+  #       path:
+  #         [
+  #           .,
+  #           tools/aliveness,
+  #           tools/hula,
+  #         ]
+  #   runs-on:
+  #     - self-hosted
+  #     - v3
+  #   container:
+  #     image: 134.28.57.223:5000/hulk
+  #     options: --user=1000:1000
+  #   steps:
+  #     - uses: actions/checkout@v3
+  #       with:
+  #         lfs: true
+  #     - name: Check
+  #       run: |
+  #         cd ${{ matrix.path }}
+  #         cargo update --frozen --workspace
```

Comment:
I don't know. Maybe @knoellle can help here. We need a solution for the patched serde crates until my PR there is merged.

Reply:
In #652 I used [`--frozen` over `--locked`](https://doc.rust-lang.org/cargo/commands/cargo-update.html#manifest-options) with `cargo update`.
I remember testing both and concluding that `--frozen` was the way to go, but don't remember why.
The documentation recommends either when checking for up to date lock files in CI, but doesn't mention potential issues like this.

However, from my testing today it seems that `cargo update --locked` does what we want.
I also tested `cargo fetch` with either flag and it seems to behave the same, except hat it also makes sure to dependencies are downloaded, but I don't think we need that.

There is one remaining issue with checking whether Cargo.lock is consistent with Cargo.toml.
When a registry patch is added to the toml without updating the lock file, cargo emits an error message like this every time it parses the dependencies:
```
warning: Patch `serde v1.0.195 (https://github.com/h3ndrk/serde.git?rev=4fec5aac7d42a4ac1fc10af037f1c85eba29fa36#4fec5aac)` was not used in the crate graph.
Patch `serde_derive v1.0.195 (https://github.com/h3ndrk/serde.git?rev=4fec5aac7d42a4ac1fc10af037f1c85eba29fa36#4fec5aac)` was not used in the crate graph.
Check that the patched package version and available features are compatible
with the dependency requirements. If the patch has a different version from
what is locked in the Cargo.lock file, run `cargo update` to use the new
version. This may also occur with an optional dependency that is not enabled.
```
I have not found a way to make this a hard error without manually parsing the stdout or rummaging through the lock file.

Reply:
It may actually matter because with `--locked` our CI might get red without us making changes because some third party was updated. Do we want this?